File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = deque()
        response = []

        if root:
            queue.append(root)

        while len(queue) > 0:
            level = []
            for i in range(len(queue)):
                element = queue.popleft()
                level.append(element.val)
                if element.left:
                    queue.append(element.left)
                if element.right:
                    queue.append(element.right)

            response.append(level)
        return response
            

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    rightSideView(node, stack = [], i=0) {
        if(!node){
            return stack
        }   
    
        if(!stack[i]){
            stack.push(node.val)
        }
    
    
        const rightStack = this.rightSideView(node.right, stack, i+1) // [1,3]
        this.rightSideView(node.left, rightStack, i+1) // [4,5,6,7,8,9]
    
        return stack
    }
}
